SparkPost/2web2ui
# AC-1612: Upgrade react-router-dom, Log all page change events to Segment

### What Changed
 - Upgraded react-router-dom to make use of `useHistory` hook
 - Created Segment helper functions to `identify` and `page`
 - On every route change (pathname change), sends a Segment `page` event.

### How To Test
 - Run 2web2ui, change pages, see events log in: https://app.segment.com/sparkpost-daniel-chalef/sources/webui_dev/debugger
 - Test other routes work correctly because of the react-router-dom upgrade
